src/charmed_kubeflow_chisme/testing/ambient_integration.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helpers for testing Istio ambient mode service mesh integration."""
+
+import aiohttp
+import lightkube
+from juju.model import Model
+
+ISTIO_K8S_APP = "istio-k8s"
+ISTIO_INGRESS_K8S_APP = "istio-ingress-k8s"
+ISTIO_BEACON_K8S_APP = "istio-beacon-k8s"
+
+
+async def deploy_and_integrate_service_mesh_charms(
+    app: str,
+    model: Model,
+    channel: str = "2/edge",
+    relate_to_ingress: bool = True,
+    relate_to_beacon: bool = True,
+) -> None:
+    """Deploy Istio service mesh charms (ambient mode).
+
+    Deploy Istio service mesh charms in the given model, relate them to the given app,
+    and wait for the model to become idle.
+
+    Args:
+        app: The name of the application to relate the Istio charms to.
+        model: The Juju model where the charms will be deployed.
+        channel: The channel from which to deploy the Istio charms. Defaults to "2/edge".
+        relate_to_ingress: Whether to integrate with the istio-ingress charm. Defaults to True.
+        relate_to_beacon: Whether to integrate with the istio-beacon charm. Defaults to True.
+    """
+    assert app in model.applications, f"application {app} was not found in model {model.name}"
```

Comment:
question: why is this required, here, if `app` is not used if not in the integration function? is it because those Istio charms will automatically detect applications in the same namespace even without integrations and so there has to be some app there?